pkg/gameservers/controller.go

```diff
@@ -936,27 +935,42 @@ func (c *Controller) syncGameServerRequestReadyState(ctx context.Context, gs *ag
 		return nil, err
 	}
 
+	gsCopy, podIPUpdated := applyGameServerPodIP(gsCopy, pod)
+	updateGSPodIP := func(gs *agonesv1.GameServer, forceUpdate bool) error {
+		if !forceUpdate {
+			return nil
+		}
+
+		_, err := c.gameServerGetter.GameServers(gs.ObjectMeta.Namespace).Update(ctx, gs, metav1.UpdateOptions{})
+		return err
+	}
```

Comment:
TODO: move to defer if `podIPUpdated`, with a `GSUpdated` check if the gs has already been updated (status changed) to avoid redoing it